src/EditorFeatures/Core/ValueTracking/ValueTrackingService.Visitor.cs

```diff
@@ -0,0 +1,266 @@
+                    }
+
+                    // If the parameter is not a ref or out param, track the reference assignments that count
+                    // as input to the argument being passed to the method.
+                    return AddReference(operation, cancellationToken);
+                }
+
+                if (IsContainedIn<IReturnOperation>(operation) || IsContainedIn<IAssignmentOperation>(operation))
+                {
+                    // If the reference is part of a return operation or assignment operation we want to track where the values come from
+                    // since they contribute to the "output" of the method/assignment and are relavent for value tracking.
+                    return AddReference(operation, cancellationToken);
+                }
+
+                return Task.CompletedTask;
+
+                Task AddReference(IOperation operation, CancellationToken cancellationToken)
+                    => operation switch
+                    {
+                        IParameterReferenceOperation parameterReference => AddOperationAsync(operation, parameterReference.Parameter, cancellationToken),
+                        IFieldReferenceOperation fieldReferenceOperation => AddOperationAsync(operation, fieldReferenceOperation.Member, cancellationToken),
+                        IPropertyReferenceOperation propertyReferenceOperation => AddOperationAsync(operation, propertyReferenceOperation.Member, cancellationToken),
+                        ILocalReferenceOperation localReferenceOperation => AddOperationAsync(operation, localReferenceOperation.Local, cancellationToken),
+                        _ => Task.CompletedTask
+                    };
+            }
+
+            private Task VisitLiteralAsync(ILiteralOperation literalOperation, CancellationToken cancellationToken)
+            {
+                if (literalOperation.Type is null)
+                {
+                    return Task.CompletedTask;
+                }
+
+                return AddOperationAsync(literalOperation, literalOperation.Type, cancellationToken);
+            }
+
+            private Task VisitReturnAsync(IReturnOperation returnOperation, CancellationToken cancellationToken)
+            {
+                if (returnOperation.ReturnedValue is null)
+                {
+                    return Task.CompletedTask;
+                }
+
+                return VisitAsync(returnOperation.ReturnedValue, cancellationToken);
+            }
+
+            private async Task AddOperationAsync(IOperation operation, ISymbol symbol, CancellationToken cancellationToken)
+            {
+                _ = await ProgressCollector.TryReportAsync(
+                        Solution,
+                        operation.Syntax.GetLocation(),
+                        symbol,
+                        cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
what does the pgrogress collector do?

Reply:
Implements `IProgress<ValueTrackedItem>`